src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -956,6 +956,53 @@ def copy_external_global(self, globalvar, tag=None):
                         " tag '{1}' is already used by another symbol."
                         "".format(globalvar, tag))
 
+    def rename_symbol(self, symbol, name):
```

Comment:
With both a "symbol not being in the symbol table" and a "name already being used" returning a `ValueError`, might this cause problems when using the method in actual code as you won't know which exception you are getting?

Reply:
That's a good point, and tbh the add method raises a KeyError if the name already exists. So this one should do the same. Changed to KeyError

Reply:
Great.